components/public-api/gitpod/experimental/v2/authprovider.proto

```diff
@@ -0,0 +1,137 @@
+
+service AuthProviderService {
+  // CreateAuthProvider creates a new auth provider.
+  rpc CreateAuthProvider(CreateAuthProviderRequest)
+      returns (CreateAuthProviderResponse) {}
+
+  // GetAuthProvider returns a single auth provider.
+  rpc GetAuthProvider(GetAuthProviderRequest)
+      returns (GetAuthProviderResponse) {}
+
+  // ListAuthProviders lists auth providers.
+  rpc ListAuthProviders(ListAuthProvidersRequest)
+      returns (ListAuthProvidersResponse) {}
+
+  // ListAuthProviderDescriptions lists publicly available descriptions of
+  // authproviders.
+  rpc ListAuthProviderDescriptions(ListAuthProviderDescriptionsRequest)
+      returns (ListAuthProviderDescriptionsResponse) {}
+
+  // UpdateAuthProvider updates an auth provider.
+  rpc UpdateAuthProvider(UpdateAuthProviderRequest)
+      returns (UpdateAuthProviderResponse) {}
+
+  // DeleteAuthProvider deletes the specified auth provider.
+  rpc DeleteAuthProvider(DeleteAuthProviderRequest)
+      returns (DeleteAuthProviderResponse) {}
+}
+
+message CreateAuthProviderRequest {
+  oneof owner {
+    string owner_id = 2;
+    string organization_id = 3;
+  }
+
+  AuthProviderType type = 4;
+  string host = 5;
+
+  OAuth2Config oauth2_config = 12;
+}
+
+message CreateAuthProviderResponse { AuthProvider auth_provider = 1; }
+
+message GetAuthProviderRequest { string auth_provider_id = 1; }
+
+message GetAuthProviderResponse { AuthProvider auth_provider = 1; }
+
+message ListAuthProvidersRequest {
+  PaginationRequest pagination = 1;
+  oneof id {
+    string user_id = 2;
+    string organization_id = 3;
+  }
+}
+
+message ListAuthProvidersResponse {
+  repeated AuthProvider list = 1;
+  PaginationResponse pagination = 2;
+}
+
+message ListAuthProviderDescriptionsRequest {
+  PaginationRequest pagination = 1;
+  oneof id {
+    string user_id = 2;
+    string organization_id = 3;
+  }
+}
+
+message ListAuthProviderDescriptionsResponse {
+  repeated AuthProviderDescription list = 1;
+  PaginationResponse pagination = 2;
+}
+
+message UpdateAuthProviderRequest {
+  string auth_provider_id = 1;
+
+  OAuth2Config oauth2_config = 12;
+}
+
+message UpdateAuthProviderResponse {}
+
+message DeleteAuthProviderRequest { string auth_provider_id = 1; }
+
+message DeleteAuthProviderResponse {}
+
+message AuthProviderDescription {
+  string id = 1;
+
+  AuthProviderType type = 4;
+  string host = 5;
+
+  optional string icon = 6;
+  optional string description = 7;
+}
+
+message AuthProvider {
+  string id = 1;
+
+  oneof owner {
+    string owner_id = 2;
+    string organization_id = 3;
+  }
+
+  AuthProviderType type = 4;
+  string host = 5;
+
+  optional string icon = 6;
+  optional string description = 7;
+  optional string settings_url = 8;
+
+  bool verified = 9;
+  bool enable_login = 10;
+
+  repeated string scopes = 11;
+
+  OAuth2Config oauth2_config = 12;
+}
+
+message OAuth2Config {
+  string client_id = 1;
+  string client_secret = 2;
+}
+
+enum AuthProviderType {
```

Comment:
is it related to SSO config somehow?

Reply:
no. the type basically identifies the strategy to be used to make it work with the specific provider type.